### PR TITLE
API v2 Improvements

### PR DESCRIPTION
**Pull request by leandrost**
Add documentation for:
- Pages (Add, replace, remove)
- Product Feeds
- Add attributes collection_id, extraction_options to create publication
- Archive Publication
- Create Collection

### CODE REVIEW

File: source/rest-v2.md [hunk 1]
```diff
@@ -667,6 +684,268 @@ The JSON response returns a list of collections with the following attributes:
...
+
+| Name                 | Type      | Required | Description
+| -------------------- | --------- | -------- | -----------------
+| title                | String    | Yes      | Collection Title
+
+# Pages
+
+## Add pages to publication
+
+Add pages from a given PDF file into a the specific publication
```
Comment (rodrigosoares) [inline source/rest-v2.md:730]: `Add pages from a given PDF file into the specific publication` (removes "a").

File: source/rest-v2.md [hunk 2]
```diff
@@ -667,6 +684,268 @@ The JSON response returns a list of collections with the following attributes:
...
+    "source_url": "https://some/file.pdf"
+  }'
+```
+
+### Request body parameters
+
+| Name                 | Type     | Required | Description
+| -------------------- | -------- | -------- | -----------
+| source_url           | String   | Yes      | URL where the PDF file resides. HTTP and HTTPS are accepted, and it needs to be a public accessible file
+| source_pages_numbers | Array    | No       | List of page numbers from pdf file to be added into the publication
```
Comment (rodrigosoares) [inline source/rest-v2.md:758]: `List of page numbers from the PDF file to be added to the publication`

File: source/rest-v2.md [hunk 3]
```diff
@@ -667,6 +684,268 @@ The JSON response returns a list of collections with the following attributes:
...
+
+`PUT https://api.publitas.com/v2/groups/<Group ID>/publications/<Publication ID>/pages/<Page Number>`
+
+### URL Parameters
+
+| Parameter        | Description
+| ---------        | --------------------------
+| Group ID         | The ID of a specific group
+| Publication ID   | The ID of a specific publication
+| Page Number      | The page number of page to be replaced
```
Comment (rodrigosoares) [inline source/rest-v2.md:780]: `The page number of a page to be replaced`

File: source/rest-v2.md [hunk 4]
```diff
@@ -667,6 +684,268 @@ The JSON response returns a list of collections with the following attributes:
...
+--data '{
+  "page": {
+    "source_url": "https://some/file.pdf"
+  }
+}'
+```
+
+### Request body parameters
+
+The following fields need to be sent within a page scope (see right for an example.)
```
Comment (rodrigosoares) [inline source/rest-v2.md:796]: `The following fields need to be sent within a page scope (see the right panel for an example):`

File: source/rest-v2.md [hunk 5]
```diff
@@ -667,6 +684,268 @@ The JSON response returns a list of collections with the following attributes:
...
+
+| Name                 | Type     | Required | Description
+| -------------------- | -------- | -------- | -----------
+| source_url           | String   | Yes      | URL where the PDF file resides. HTTP and HTTPS are accepted, and it needs to be a public accessible file
+| source_pages_number  | Integer  | No       | PDF page number to be used on replace (by default page number 1 is used)
+| extraction_options   | Object   | No       | Define settings to extract hotspots from the PDF file. See [extraction options](#extraction-options) for allowed fields
+| remove_hotspots      | Boolean  | No       | If true remove current hotspots on page to be replaced
```
Comment (rodrigosoares) [inline source/rest-v2.md:803]: `If "true", remove current hotspots from the page to be replaced`

File: source/rest-v2.md [hunk 6]
```diff
@@ -667,6 +684,268 @@ The JSON response returns a list of collections with the following attributes:
...
+curl -H "Authorization: ApiKey <api_key>" -X DELETE "https://api.publitas.com/v2/groups/1/publications/222"
+```
+
+### URL Parameters
+
+| Parameter        | Description
+| ---------        | --------------------------
+| Group ID         | The ID of a specific group
+| Publication ID   | The ID of a specific publication
+| Page Number      | The page number to be removed
```
Comment (rodrigosoares) [inline source/rest-v2.md:825]: `The number of the page to be removed`

File: source/rest-v2.md [hunk 7]
```diff
@@ -667,6 +684,268 @@ The JSON response returns a list of collections with the following attributes:
...
+
+### Response codes
+
+This endpoint returns the `200` response code.
+
+# Product Feeds
+
+## List recent product feeds
+
+List last 5 product feed imports
```
Comment (rodrigosoares) [inline source/rest-v2.md:836]: `List the last 5 product feed imports` (imperative) or `Lists the last 5 product feed imports` (what it does).

File: source/rest-v2.md [hunk 8]
```diff
@@ -667,6 +684,268 @@ The JSON response returns a list of collections with the following attributes:
...
+
+```shell
+curl --location "http://api.publitas.com/v2/groups/1/product_feeds" --header "Authorization: ApiKey <api_key>"
+```
+
+> The above command returns a JSON document structured like so:
```
Comment (rodrigosoares) [inline source/rest-v2.md:842]: `The above command returns a JSON document structured like this:`

File: source/rest-v2.md [hunk 9]
```diff
@@ -667,6 +684,268 @@ The JSON response returns a list of collections with the following attributes:
...
+
+### URL Parameters
+
+| Parameter        | Description
+| ---------        | --------------------------
+| Group ID         | The ID of a specific group
+
+### Request body parameters
+
+The following fields need to be sent within a product feed scope (see right for an example.)
```
Comment (rodrigosoares) [inline source/rest-v2.md:943]: `The following fields need to be sent within a product feed scope (see the right panel for an example):`

File: source/rest-v2.md [hunk 10]
```diff
@@ -697,6 +976,21 @@ The JSON response returns a list of collections with the following attributes:
 | Turkish    | tr   |
 | Ukranian   | uk   |
 
+# Extraction Options
+
+The following fields defines options to extract hotspots from a PDF
```
Comment (rodrigosoares) [inline source/rest-v2.md:981]: `The following fields define options to extract hotspots from a PDF`

File: source/rest-v2.md [hunk 11]
```diff
@@ -697,6 +976,21 @@ The JSON response returns a list of collections with the following attributes:
...
+
+| Name                | Type   | Required | Description                                                                                                                                       |
+| ------------------- | ------ | -------- | ------------------------------------------------------------------------------------------------------------------------------------------------- |
+| links               | Object | No       | Extracts link hotspots for any URLs, document links, and email addresses found in PDF file (see object definition table bellow)                   |
```
Comment (rodrigosoares) [inline source/rest-v2.md:985]: `Extracts link hotspots from any URLs, document links, and email addresses found in the PDF file (see object definition table below)`

File: source/rest-v2.md [hunk 12]
```diff
@@ -697,6 +976,21 @@ The JSON response returns a list of collections with the following attributes:
...
+| products            | Object | No       | Extracts product hotspots from any SKU found in PDF file, matching product feed or product library set (see object definition table bellow)       |
```
Comment (rodrigosoares) [inline source/rest-v2.md:986]: `Extracts product hotspots from any SKUs found in PDF file, matching the product feed or the product library set (see object definition table below)`

File: source/rest-v2.md [hunk 13]
```diff
@@ -697,6 +976,21 @@ The JSON response returns a list of collections with the following attributes:
...
+
+Both 'links' and 'products' allow the fields bellow:
```
Comment (rodrigosoares) [inline source/rest-v2.md:988]: `Both 'links' and 'products' allow the fields below:`

File: source/rest-v2.md [hunk 14]
```diff
@@ -697,6 +976,21 @@ The JSON response returns a list of collections with the following attributes:
...
+
+| Name                | Type     | Required | Description                                                                                                                              |
+| ------------------- | -------- | -------- | -----------------------------------------------------------------------------------------------------------------------------------------|
+| icons               | Boolean  | No       | Show/hide hotspot icon indication, is not applicable hotspot extracted from pdf annotations(those are defined in the annotantion itself) |
```
Comment (rodrigosoares) [inline source/rest-v2.md:992]: `Show/hide hotspot icon indication. It is not applicable if the hotspot is extracted from PDF annotations (those are defined in the annotation itself)`

File: source/rest-v2.md [hunk 15]
```diff
@@ -667,6 +684,268 @@ The JSON response returns a list of collections with the following attributes:
...
+    "collection": {
+      "id": 42,
+      "title": "New Collection"
+    }
+  }
+```
+
+### Request body parameters
+
+The following fields need to be sent within a collection scope (see right for an example.)
```
Comment (rodrigosoares) [inline source/rest-v2.md:720]: `The following fields need to be sent within a collection scope (see the right panel for an example):`

**Review by rodrigosoares — APPROVED** [review]
Overall all good. Just left some grammar suggestions, as this doc is exposed to customers.

**Comment by felixhageloh** [comment]
I thought about this today: I think it might be better to only update the docs once Dan has used this API for a bit. This way we can still easily change some details if need be. We can finalize this PR (if not already) and then merge in a few weeks.

**Pull request merged**